### ISSUE TITLE
Add events on the PathResolving process to implement redirects from other plugins. 

### ISSUE THREAD

**Issue by JuicyLung91** [elkmod/SwagShopwarePwa#73]
There should be a possibility to change the url when a path is resolved and the api response should have a property that indicates a redirect. 

Thinking about something like this: 
```
{
    resourceType: 'frontend.navigation.redirect',
    redirectTarget: '/target-of-redirect',
    redirectStatusCode: 302
}
```

**Comment by JuicyLung91** [comment]
I made a PR for this. What do you think? 
With this you can add two events from the plugin: 

Check if a redirect exists before the the PathResolving process and add the new path. 

```

public function beforeUrlResolved(BeforeSeoResolverResolveEvent $event): void {
    $requestedUrl = $event->getPath();
    if ($requestedUrl && ($requestedUrl === '' || $requestedUrl === '/') ) {
        return;
    }

    $redirect = $this->findRedirect($requestedUrl, $event->getSalesChannelContext()->getContext()); //find redirect and set it
    if ($redirect) {
        $event->setPath($redirect->getUrl());
    }
}
```

Check if the original request url is the same as the actual url. If it differs, check if a redirect exists for the original url and add the necessary data to end pageLoader  so the pwa can react to this. 
```
    public function onPageLoaderLoaded(PageLoaderLoadedEvent $pageLoaderLoadedEvent) {
        $pageLoaderResult = $pageLoaderLoadedEvent->getPageLoaderResult();
        $request = $pageLoaderLoadedEvent->getRequest();
        $requestPath = $request->get('path');
        $loadedPath = $pageLoaderLoadedEvent->getCanonicalPathInfo();

        if ($loadedPath !== $requestPath) {
            $redirect = $this->findRedirect($requestPath, $pageLoaderLoadedEvent->getPageLoaderContext()->getContext()->getContext());

            $pageLoaderResult->addExtension(
                'seo_redirect', new ArrayStruct(
                    [
                        'oldUrl' => $redirect->getOld(),
                        'redirectTarget' => $redirect->getUrl(),
                        'redirectStatusCode' => 301
                    ],
                    'api_seo_redirect'
                )
            );
        }

    }
```
I found this very handy when you migrate from a different system to sw6 and you have multiple redirects. i.e in a csv file.

**Comment by elkmod** [comment]
Thanks for the issue and the contribution, @JuicyLung91

We're having a look and will get back to you soon!

**Comment by JuicyLung91** [comment]
I'd  like to reopen this issue. Since I didn't have time to make the adjustments in the last PR and some things have changed in the meantime, I have slightly changed things for the new version. Also, the request passing that you mentioned [here](https://github.com/shopware/SwagShopwarePwa/pull/75 ), I have adapted and do not pass the request at all. Only within the PageLoaderLoadedEvent class I return the request in the getter from the pageLoaderContext back. Since here the interface is already in use (on my projects). 

Since the last PR is closed I can not push into it. And didn´t find out how to reopen it. I created a new PR: https://github.com/shopware/SwagShopwarePwa/pull/101
